File: barber-app/backend/src/entities/Barbershop.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  CreateDateColumn,
  UpdateDateColumn,
  OneToMany,
} from 'typeorm';
import { Barber } from './Barber';
import { Service } from './Service';

@Entity('barbershops')
export class Barbershop {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column()
  name: string;

  @Column({ nullable: true })
  description: string;

  @Column()
  address: string;

  @Column({ nullable: true })
  phone: string;

  @Column({ nullable: true })
  logo: string;

  @Column({ type: 'time', default: '09:00:00' })
  openTime: string;

  @Column({ type: 'time', default: '18:00:00' })
  closeTime: string;

  @Column({ default: true })
  isActive: boolean;

  @OneToMany(() => Barber, (barber) => barber.barbershop)
  barbers: Barber[];

  @OneToMany(() => Service, (service) => service.barbershop)
  services: Service[];

  @CreateDateColumn()
  createdAt: Date;

  @UpdateDateColumn()
  updatedAt: Date;
}

File: barber-app/barber-app/src/main/java/com/barber/domain/entity/Barbershop.java
package com.barber.domain.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "barbershops")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Barbershop {
    
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;
    
    @Column(nullable = false)
    private String name;
    
    @Column(nullable = false)
    private String address;
    
    private String phone;
    
    @Column(columnDefinition = "TEXT")
    private String description;
    
    @Column(name = "open_time")
    private String openTime;
    
    @Column(name = "close_time")
    private String closeTime;
    
    @Column(name = "is_active", nullable = false)
    private Boolean isActive;
    
    @PrePersist
    protected void onCreate() {
        if (isActive == null) {
            isActive = true;
        }
    }
}
